tests/core/test_envs_manager.py

```diff
@@ -112,3 +113,30 @@ def test_list_all_known_prefixes_with_permission_error(mock_clean_env, mock_get_
     all_env_paths = list_all_known_prefixes()
     # On Windows, all_env_paths can contain more paths (like '\\Miniconda')
     assert "root_prefix" in all_env_paths
+
+
+@pytest.mark.skipif(on_win, reason="test is invalid on windows")
+@patch("conda.core.envs_manager.context")
+@patch("conda.core.envs_manager._clean_environments_txt")
+@patch("pwd.getpwall")
+@patch("conda.core.envs_manager.is_admin")
+def test_list_all_known_prefixes_with_none_values_error(
+    mock_is_admin, mock_getpwall, mock_clean_env, mock_context, tmp_path
+):
+    """
+    Regression test for a bug first indentified in this issue: https://github.com/conda/conda/issues/12063
+
+    Tests to make sure that `None` values are filtered out of the `search_dirs` variable in the
+    `list_all_known_prefixes` function.
+    """
+    mock_is_admin.return_value = True
+    mock_getpwall.return_value = [Namespace(pw_dir=expand("~")), Namespace(pw_dir=None)]
```

Comment:
I would use something besides Namespace here, either an inline class or a dataclass or namedtuple, to avoid wondering if this had anything to do with argparse